### ISSUE TITLE
Windows install

### ISSUE THREAD

**Issue by skupu** [zfortier/arduino_keyboard_emulator#1]
Hi,

Unofortunately I couldn't manage to get it working - it's not easy for Windows users...

Could you make a detailed description for Windows?

Is this board suitable: https://allegro.pl/oferta/atmega328-mikrokomputer-uno-ch340-1-szt-14066391989 ?

**Comment by zfortier** [comment]
> Hi,
> 
> Unofortunately I couldn't manage to get it working - it's not easy for Windows users...
> 
> Could you make a detailed description for Windows?
> 
> Is this board suitable: https://allegro.pl/oferta/atmega328-mikrokomputer-uno-ch340-1-szt-14066391989 ?

Yes, that board is exactly the one I used for building and testing, it will work perfectly. 

Unfortunately I don't have access to any Windows systems, so I can't give instructions on getting it to work there, but I'll give a few thoughts:

  - If you have Bash running (newer versions of Windows should allow to run Bash natively, or use Cygwin on older versions), then you can use the scripts.
  - You'll need an appropriate cross compiler installed (which you should have anyway if you are using the Uno board)
  - This is using a custom Makefile and not the Arduino IDE, so you also need to have something like GNU make that can use that file to perform the build
  - you'll need `dfu-programmer` to use the scripts. Even without the scripts it is required to have some way to update the device firmware on the Uno, since this is overwriting the Arduino boot loader with one that tells the host OS that it's a keyboard.

If you are getting stuck using the Makefile, you can get around that by moving some code around. I haven't tried this, but what would probably work is taking the content of the header files (in the `include` directory) and the source file (in the `src` directory) and pasting it all into a single .ino file through the Arduino IDE. Then replace the main function by taking everything before the `while` loop and pasting it into `setup`, and taking the content of the `while` loop and pasting it into `loop`.
